getEtcdClusterMembers - index out of range [3] with length 3

**Describe the bug**
getEtcdClusterMembers throws "index out of range [3] with length 3" error when trying to get etcd members. 

**To Reproduce**
Steps to reproduce the behavior:

pgwatch: version 3.7.0
OS: Rocky Linux 8.10 Linux 4.18.0-553.el8_10.x86_64

podman run -d --name pgwatch-agent \
  -p 9187:9187 \
  -p 8080:8080 \
  -v /opt/pgwatch/sources.yaml:/sources.yaml \
  cybertecpostgresql/pgwatch:latest \
  --sources=/sources.yaml \
  --sink=postgresql://pgwatch:*******@pg_watch:5432/measurements

source.yaml

- name: app
  kind: patroni
  preset_metrics: exhaustive
  is_enabled: true
  host_config:
    dcs_type: "etcd"
    dcs_endpoints: ["http://pgdb02:2379"]
    scope: "szkolenie"
    namespace: "/service/"
    username: ""
    password: ""
    ca_file: ""
    cert_file: ""
    key_file: ""
    logs_glob_path: ""
    logs_match_regex: ""
    per_metric_disabled_intervals: []
  only_if_master: false

[postgres@pgdb01:~]# patronictl version szkolenie
patronictl version 4.0.6

pgdb01: Patroni 4.0.6 PostgreSQL 17.6
pgdb02: Patroni 4.0.6 PostgreSQL 17.6
pgdb03: Patroni 4.0.6 PostgreSQL 17.6

patronictl -d etcd3://pgdb02:2379 list szkolenie
+ Cluster: szkolenie (7524870463057937086) --+----+-----------+--------------+
| Member |  Host  |     Role     |   State   | TL | Lag in MB |     Tags     |
+--------+--------+--------------+-----------+----+-----------+--------------+
| pgdb01 | pgdb01 | Leader       | running   | 33 |           |              |
| pgdb02 | pgdb02 | Sync Standby | streaming | 33 |         0 |              |
| pgdb03 | pgdb03 | Replica      | streaming | 33 |         0 | nosync: true |
+--------+--------+--------------+-----------+----+-----------+--------------+

etcd: 
etcdctl version
etcdctl version: 3.6.1
API version: 3.6

**Screenshots**
2025-09-20 22:41:56.716 [INFO] [sink:postgres] [db:measurements] initialising measurements database...
2025-09-20 22:41:56.719 [INFO] [sink:postgres] [db:measurements] measurements sink is activated
2025-09-20 22:41:56.727 [ERROR] [callstack:goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:26 +0x5e
main.main.func1()
	/pgwatch/cmd/pgwatch/main.go:51 +0xd0
panic({0x10080a0?, 0xc0000426d8?})
	/usr/local/go/src/runtime/panic.go:792 +0x132
github.com/cybertec-postgresql/pgwatch/v3/internal/sources.getEtcdClusterMembers({{0xc0000105bc, 0x3}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, {0xc0000105e0, 0x7}, ...})
	/pgwatch/internal/sources/resolver.go:181 +0xcc5
github.com/cybertec-postgresql/pgwatch/v3/internal/sources.ResolveDatabasesFromPatroni({{0xc0000105bc, 0x3}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, {0xc0000105e0, 0x7}, ...})
	/pgwatch/internal/sources/resolver.go:203 +0x78
github.com/cybertec-postgresql/pgwatch/v3/internal/sources.Source.ResolveDatabases({{0xc0000105bc, 0x3}, {0x0, 0x0}, {0x0, 0x0}, 0x0, 0x0, {0xc0000105e0, 0x7}, ...})
	/pgwatch/internal/sources/resolver.go:46 +0x158
github.com/cybertec-postgresql/pgwatch/v3/internal/sources.Sources.ResolveDatabases({0xc0001ede00, 0x1, 0x1?})
	/pgwatch/internal/sources/resolver.go:35 +0x138
github.com/cybertec-postgresql/pgwatch/v3/internal/reaper.(*Reaper).LoadSources(0xc00033a7e0)
	/pgwatch/internal/reaper/reaper.go:384 +0x17e
github.com/cybertec-postgresql/pgwatch/v3/internal/reaper.(*Reaper).Reap(0xc00033a7e0, {0x17d15b0, 0xc000305dd0})
	/pgwatch/internal/reaper/reaper.go:83 +0x19b
main.main()
	/pgwatch/cmd/pgwatch/main.go:110 +0x685

--- a/internal/sources/resolver.go
+++ b/internal/sources/resolver.go
@@ -57,6 +57,10 @@ type PatroniClusterMember struct {
 	Role    string
 }
 
+func (pcm PatroniClusterMember) IsPrimary() bool {
+	return pcm.Role == "primary" || pcm.Role == "master"
+}
+
 var logger log.Logger = log.FallbackLogger
 
 var lastFoundClusterMembers = make(map[string][]PatroniClusterMember) // needed for cases where DCS is temporarily down
@@ -171,8 +175,8 @@ func getEtcdClusterMembers(s Source, hc HostConfig) ([]PatroniClusterMember, err
 		}
 		// remove leading slash and split by "/"
 		parts := strings.Split(strings.TrimPrefix(string(node.Key), "/"), "/")
-		if len(parts) < 3 {
-			return nil, errors.New("invalid ETCD key format")
+		if len(parts) < 4 || parts[2] != "members" {
+			continue // skip non-member keys
 		}
 		role := nodeData["role"]
 		connURL := nodeData["conn_url"]
@@ -281,12 +285,12 @@ func ResolveDatabasesFromPatroni(source Source) (SourceConns, error) {
 
 	for _, patroniMember := range clusterMembers {
 		logger.Info("Processing Patroni cluster member: ", patroniMember.Name)
-		if source.OnlyIfMaster && patroniMember.Role != "master" {
+		if source.OnlyIfMaster && !patroniMember.IsPrimary() {
 			continue
 		}
 		src := *source.Clone()
 		src.ConnStr = patroniMember.ConnURL
-		if hostConfig.IsScopeSpecified() {
+		if !hostConfig.IsScopeSpecified() {
 			src.Name += "_" + patroniMember.Scope
 		}
 		src.Name += "_" + patroniMember.Name

--- a/internal/sources/resolver_test.go
+++ b/internal/sources/resolver_test.go
@@ -3,7 +3,6 @@ package sources_test
 import (
 	"context"
 	"errors"
-	"fmt"
 	"strings"
 	"testing"
 	"time"
@@ -54,6 +53,7 @@ func TestMonitoredDatabase_ResolveDatabasesFromPostgres(t *testing.T) {
 }
 
 func TestMonitoredDatabase_ResolveDatabasesFromPatroni(t *testing.T) {
+
 	etcdContainer, err := etcd.Run(ctx, "gcr.io/etcd-development/etcd:v3.5.14",
 		testcontainers.WithWaitStrategy(wait.ForLog("ready to serve client requests").
 			WithStartupTimeout(15*time.Second)))
@@ -82,18 +82,24 @@ func TestMonitoredDatabase_ResolveDatabasesFromPatroni(t *testing.T) {
 	)
 	require.NoError(t, err)
 	defer func() { assert.NoError(t, pgContainer.Terminate(ctx)) }()
-
+	pgConnStr, err := pgContainer.ConnectionString(ctx, "sslmode=disable")
+	require.NoError(t, err)
 	// Put values to etcd server
-	cancelCtx, cancel := context.WithTimeout(context.Background(), time.Second)
-	defer cancel()
-	connStr, err := pgContainer.ConnectionString(cancelCtx, "sslmode=disable")
-	require.NoError(t, err)
-	_, err = cli.Put(cancelCtx, "/service/batman/members/pg1",
-		fmt.Sprintf(`{"role":"master","conn_url":"%s"}`, connStr))
-	require.NoError(t, err)
-	_, err = cli.Put(cancelCtx, "/service/batman/members/pg2",
-		`{"role":"standby","conn_url":"must_be_skipped"}`)
-	require.NoError(t, err)
+	kv := map[string]string{
+		`/service/demo/config`:           `{"ttl":30,"loop_wait":10,"retry_timeout":10,"maximum_lag_on_failover":1048576,"postgresql":{"use_pg_rewind":true,"pg_hba":["local all all trust","host replication replicator all md5","host all all all md5"],"parameters":{"max_connections":100}}}`,
+		`/service/demo/initialize`:       `7553211779477532695`,
+		`/service/demo/leader`:           `patroni3`,
+		`/service/demo/members/patroni1`: `{"conn_url":"postgres://172.18.0.8:5432/postgres","api_url":"http://172.18.0.8:8008/patroni","state":"running","role":"replica","version":"4.0.7","xlog_location":67108960,"replay_lsn":67108960,"receive_lsn":67108960,"replication_state":"streaming","timeline":1}`,
+		`/service/demo/members/patroni2`: `{"conn_url":"postgres://172.18.0.4:5432/postgres","api_url":"http://172.18.0.4:8008/patroni","state":"running","role":"replica","version":"4.0.7","xlog_location":67108960,"replay_lsn":67108960,"receive_lsn":67108960,"replication_state":"streaming","timeline":1}`,
+		`/service/demo/members/patroni3`: `{"conn_url":"` + pgConnStr + `","api_url":"http://172.18.0.3:8008/patroni","state":"running","role":"primary","version":"4.0.7","xlog_location":67108960,"timeline":1}`,
+		`/service/demo/status`:           `{"optime":67108960,"slots":{"patroni1":67108960,"patroni2":67108960,"patroni3":67108960},"retain_slots":["patroni1","patroni2","patroni3"]}}`}
+
+	cancelCtx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	for k, v := range kv {
+		_, err = cli.Put(cancelCtx, k, v)
+		require.NoError(t, err, "failed to put key %s to etcd", k)
+	}
+	cancel()
 
 	md := sources.Source{}
 	md.Name = "continuous"
@@ -103,7 +109,7 @@ func TestMonitoredDatabase_ResolveDatabasesFromPatroni(t *testing.T) {
 		md.Kind = sources.SourcePatroni
 		md.ConnStr = "etcd://" + strings.TrimPrefix(endpoint, "http://")
 		md.ConnStr += "/service"
-		md.ConnStr += "/batman"
+		md.ConnStr += "/demo"
 
 		// Run ResolveDatabasesFromPatroni
 		dbs, err := md.ResolveDatabases()
@@ -117,7 +123,7 @@ func TestMonitoredDatabase_ResolveDatabasesFromPatroni(t *testing.T) {
 		e := strings.TrimPrefix(endpoint, "http://")
 		md.ConnStr = "etcd://" + strings.Join([]string{e, e, e}, ",")
 		md.ConnStr += "/service"
-		md.ConnStr += "/batman"
+		md.ConnStr += "/demo"
 
 		// Run ResolveDatabasesFromPatroni
 		dbs, err := md.ResolveDatabases()

